HopsanCore/src/ComponentSystem.cpp

```diff
@@ -882,6 +882,24 @@ std::vector<HString> ComponentSystem::getSubComponentNames() const
     return names;
 }
 
+
+//! @brief Returns a list of pointers to all components the system and all sub systems
+//! @param [out] rComponents Pointers to all components
+void ComponentSystem::getSubComponentsRecursively(std::vector<Component*> &rComponents) const
+{
+    //! @todo for now create a vector of the component names, later maybe we should return a pointer to the real internal map
+    SubComponentMapT::const_iterator it;
+    for (it = mSubComponentMap.begin(); it != mSubComponentMap.end(); ++it)
+    {
+        if(it->second->isComponentSystem()) {
+            reinterpret_cast<ComponentSystem*>(it->second)->getSubComponentsRecursively(rComponents);
```

Comment:
This will not include the system itself, but I assume it should be included since it is also a component, which may have ports.